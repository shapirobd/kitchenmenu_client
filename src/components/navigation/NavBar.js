import React, { useState } from "react";
import { Link, useHistory, useLocation } from "react-router-dom";
import {
	AppBar,
	Toolbar,
	Button,
	InputBase,
	IconButton,
} from "@material-ui/core";
import SearchIcon from "@material-ui/icons/Search";
import MenuIcon from "@material-ui/icons/Menu";
import { useStyles } from "./styles/NavBarStyles";
import logo from "../../images/logo.png";
import { useSelector, useDispatch } from "react-redux";
import { logout } from "../../actionCreators/userActionCreators";
import { loadFeed } from "../../actionCreators/recipeActionCreators";
import useWindowDimensions from "../../customHooks/getWindowDimensions";
import SideNavDrawer from "./SideNavDrawer";

const NavBar = () => {
	const classes = useStyles();
	const dispatch = useDispatch();
	const history = useHistory();
	const location = useLocation();
	const { width } = useWindowDimensions();

	const user = useSelector((state) => state.user);

	const handleLogout = (evt) => {
		evt.preventDefault();
		dispatch(logout());
		history.push("/");
	};

	const handleChange = (evt) => {
		setSearchData(evt.target.value);
	};

	const handleSearch = (evt) => {
		evt.preventDefault();
		dispatch(loadFeed(1, searchData));
		if (location.pathname !== "/") {
			history.push("/");
		}
	};

	const [state, setState] = React.useState({
		left: false,
	});

	const toggleDrawer = (anchor, open) => (event) => {
		if (
			event &&
			event.type === "keydown" &&
			(event.key === "Tab" || event.key === "Shift")
		) {
			return;
		}
		setState({ ...state, [anchor]: open });
	};

	const [searchData, setSearchData] = useState("");

	return (
		<div className={classes.root}>
			{console.log(width)}
			<AppBar position="fixed" className={classes.appBar}>
				<Toolbar classes={{ root: classes.root }}>
					{width <= 599 && user ? (
						<>
							<IconButton
								edge="start"
								className={classes.menuButton}
								color="inherit"
								aria-label="open drawer"
								onClick={toggleDrawer("left", true)}
							>
								<MenuIcon />
							</IconButton>
							<SideNavDrawer
								state={state}
								toggleDrawer={toggleDrawer}
								user={user}
								mobile={true}
							/>
						</>
					) : null}
					{width > 599 ? (
						<Link to="/" className={`${classes.navLink} ${classes.title}`}>
							<img className={classes.logo} src={logo} alt={logo} />
						</Link>
					) : null}

					<form
						className={width <= 599 ? classes.searchMobile : classes.search}
						onSubmit={handleSearch}
					>
						<div className={classes.searchIcon}>
							<SearchIcon />
						</div>
						<InputBase
							placeholder="Searchâ€¦"
							classes={{
								root: classes.inputRoot,
								input: classes.input,
							}}
							inputProps={{ "aria-label": "search" }}
							onChange={handleChange}
						/>
					</form>
					<div
						style={
							width > 599
								? { float: "right" }
								: {
										width: "50%",
										float: "right",
										display: "flex",
										flexDirection: "row",
										justifyContent: "flex-end",
								  }
						}
					>
						{user ? (
							<>
								<Button
									onClick={handleLogout}
									color="inherit"
									className={classes.navLink}
								>
									Logout
								</Button>
							</>
						) : (
							<>
								<Link
									to="/login"
									className={classes.navLink}
									// style={{ width: "50%" }}
								>
									<Button color="inherit">Login</Button>
								</Link>
								<Link
									to="/signup"
									className={classes.navLink}
									// style={{ width: "50%" }}
								>
									<Button color="inherit" className={classes.signupBtn}>
										Sign up
									</Button>
								</Link>
							</>
						)}
					</div>
				</Toolbar>
			</AppBar>
		</div>
	);
};

export default NavBar;
