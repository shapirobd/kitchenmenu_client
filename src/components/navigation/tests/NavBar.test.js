import React from "react";
import { render, screen } from "@testing-library/react";
import NavBar from "../NavBar";
import { MemoryRouter } from "react-router-dom";
import { PersistGate } from "redux-persist/integration/react";
import { Provider } from "react-redux";
import { store, persistor } from "../../../configureStore";

it("should render without crashing", () => {
	render(
		<Provider store={store}>
			<MemoryRouter>
				<PersistGate loading={null} persistor={persistor}>
					<NavBar />
				</PersistGate>
			</MemoryRouter>
		</Provider>
	);
});

it("should match screenshot", () => {
	const { asFragment } = render(
		<Provider store={store}>
			<MemoryRouter>
				<PersistGate loading={null} persistor={persistor}>
					<NavBar />
				</PersistGate>
			</MemoryRouter>
		</Provider>
	);

	expect(asFragment()).toMatchSnapshot();
});
