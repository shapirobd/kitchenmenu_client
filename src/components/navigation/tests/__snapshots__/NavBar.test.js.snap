// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match screenshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-13"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-14 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root makeStyles-root-13 MuiToolbar-regular MuiToolbar-gutters"
      >
        <a
          class="makeStyles-navLink-18 makeStyles-title-16"
          href="/"
        >
          <img
            alt="logo.png"
            class="makeStyles-logo-17"
            src="logo.png"
          />
        </a>
        <form
          class="makeStyles-search-20"
        >
          <div
            class="makeStyles-searchIcon-22"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="MuiInputBase-root makeStyles-inputRoot-23"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input makeStyles-input-24"
              placeholder="Searchâ€¦"
              type="text"
              value=""
            />
          </div>
        </form>
        <div
          style="float: right;"
        >
          <a
            class="makeStyles-navLink-18"
            href="/login"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Login
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </a>
          <a
            class="makeStyles-navLink-18"
            href="/signup"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-signupBtn-19 MuiButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Sign up
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
